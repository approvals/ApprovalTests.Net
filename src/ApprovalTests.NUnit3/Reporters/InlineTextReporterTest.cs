[TestFixture]
public class InlineTextReporterTest
{
    [Test]
    public void TestComment()
    {
        var actual = "Hello";
        var expected = "Hello";
        Approvals.AssertText(expected, actual);
    }

    [Test]
    public void CSharpStrings()
    {
        var example1 = new[]
        {
            "**************",

            "I am ten chars",
            "**************",
        };
        var samples = new[] { "boo","with \"quotes\".", example1.JoinWith("\n"), };

        Approvals.VerifyAll("C# Strings", samples, InlineTextReporter.ConvertToCSharp);
    }

    [Test]
    public void MakerHeadertest()
    {
        // begin-snippet: assert_text
        var header = new Header();
        var actual = header.MakeHeading("I am ten chars");
        var expected = new[]{
            "**************",
            "I am ten chars",
            "**************",

        };
        Approvals.AssertText(expected, actual);
        // end-snippet
    }

    public void MakerHeaderBefore()
    {
        // begin-snippet: assert_text_before
        var header = new Header();
        var actual = header.MakeHeading("I am ten chars");
        var expected = "";
        Approvals.AssertText(expected, actual);
        // end-snippet
    }
}

public class Header
{
    public string MakeHeading(string iAmTenChars) =>
        new[]
        {
            "**************",

            "I am ten chars",
            "**************",
        }.JoinWith("\n");
}