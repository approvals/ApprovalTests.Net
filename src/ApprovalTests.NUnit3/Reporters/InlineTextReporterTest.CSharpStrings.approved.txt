C# Strings

var expected = "boo";
var expected = "with \"quotes\".";
                var expected = new[]{
                "**************",
                "I am ten chars",
                "**************",
                };
